File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Coolstore/Models/CoolstoreOrderItem.cs
namespace Ecommerce.Coolstore.Models;

public class CoolstoreOrderItem
{
    public string ProductId { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public double Price { get; set; }

    public CoolstoreOrderItem()
    {
    }

    public CoolstoreOrderItem(string productId, int quantity, double price)
    {
        ProductId = productId;
        Quantity = quantity;
        Price = price;
    }
}

File: rgctl-tests/ecommerce-c/src/coolstore/models/coolstore_order_item.c
#include "ecommerce/coolstore/models/coolstore_order_item.h"
#include <string.h>

void coolstore_order_item_init(coolstore_order_item_t *item, const char *productId,
                               int quantity, double price) {
    if (!item) return;
    memset(item, 0, sizeof(*item));
    if (productId) strncpy(item->productId, productId, sizeof(item->productId) - 1);
    item->quantity = quantity;
    item->price = price;
}
